### PR TITLE
Load models from URL for issue 579

### PR DESCRIPTION
**Pull request by dselman**
Signed-off-by: Daniel Selman <danscode@selman.org>

## Checklist
 - [x]  A link to the issue/user story that the pull request relates to
 - [x]  How to recreate the problem without the fix
 - [x]  Design of the fix
 - [x]  How to prove that the fix works
 - [x]  Automated tests that prove the fix keeps on working
 - [x]  Documentation - any JSDoc, website, or Stackoverflow answers?

## Issue/User story
#579 

## Steps to Reproduce

1.
2.
3.
4.

## Existing issues

- [x] [Stack Overflow issues](http://stackoverflow.com/tags/hyperledger-composer)
- [x] [GitHub Issues](https://github.com/hyperledger/composer/issues)
- [x] [Rocket Chat history](https://chat.hyperledger.org/channel/composer)

### CODE REVIEW

File: packages/composer-common/lib/introspect/loaders/jobqueue.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    /* eslint-disable class-methods-use-this */
+
+    /* istanbul ignore next */
+    /**
+     * Abstract method: must be overriden.
+     * @param {any} job - data for the job
+     * @return {Promise} promise to the job results.
+     */
+    runJob(job) { return Promise.resolve(job); }
```
Comment (sstone1) [inline packages/composer-common/lib/introspect/loaders/jobqueue.js]: I think if you add the @abstract tag to the JSDoc comment you don't need any of these eslint/istanbul hacks 😉

File: packages/composer-common/package.json [hunk 2]
```diff
@@ -49,12 +49,14 @@
...
     "mocha": "3.4.2",
     "mockery": "2.0.0",
+    "moxios": "0.4.0",
     "nyc": "11.1.0",
     "pegjs": "0.9.0",
     "sinon": "2.3.8"
   },
   "dependencies": {
     "acorn": "5.1.2",
+    "axios": "0.17.1",
```
Comment (sstone1) [inline packages/composer-common/package.json]: Do we really need yet another HTTP request module? We already use request and xhr.

Comment (dselman) [reply packages/composer-common/package.json]: I initially wrote the code using request, but I couldn't convince it to webpack (and it was huge). I wanted something that would potentially run in Playground so we can add an "update models" button do the line if necessary.

Also the moxios library makes it really easy to mock the calls to axios... which is nice.

File: packages/composer-common/lib/introspect/loaders/jobqueue.js (hunk 1 above)
Comment (dselman) [reply packages/composer-common/lib/introspect/loaders/jobqueue.js]: I will make that @abstract

**Comment by dselman** [comment]
@sstone1 eslint issue is resolved, answered the Axios question. The use of Axios is fairly well encapsulated, so easy to swap out in the future if necessary.

File: packages/composer-common/package.json (hunk 2 above)
Comment (sstone1) [reply packages/composer-common/package.json]: I think I managed to get `request` working in webpack and get rid of `xhr`. I'm not super fussed as the dream is to get rid of `request` altogether once we support external requires, and if `moxios` makes our lives easier from a testing point of view then we could move all the internal HTTP code over to that.

**Pull request merged**